client/app/pages/queries/hooks/useAddNewParameterDialog.js

```diff
@@ -0,0 +1,24 @@
+import { isFunction, map } from "lodash";
+import { useCallback, useRef } from "react";
+import EditParameterSettingsDialog from "@/components/EditParameterSettingsDialog";
+
+export default function useAddNewParameterDialog(query, onParameterAdded) {
+  const onParameterAddedRef = useRef(null);
+  onParameterAddedRef.current = isFunction(onParameterAdded) ? onParameterAdded : () => {};
```

Comment:
Confirming: is this ref to make sure the most recent `onParameterAdded` is used in the promise?

Reply:
Yes, exactly